CopilotKit/packages/runtime/src/graphql/resolvers/copilot.resolver.ts

```diff
@@ -73,22 +188,48 @@ export class CopilotResolver {
 
                 // signal when we are done streaming
                 const streamingTextStatus = new Subject<typeof MessageStatusUnion>();
-
                 // push the new message
                 pushMessage({
                   id: nanoid(),
                   status: firstValueFrom(streamingTextStatus),
                   createdAt: new Date(),
                   role: MessageRole.assistant,
                   content: new Repeater(async (pushTextChunk, stopStreamingText) => {
-                    // push the message content
-                    await textMessageContentStream.forEach(async (e: RuntimeEvent) => {
-                      if (e.type == RuntimeEventTypes.TextMessageContent) {
-                        await pushTextChunk(e.content);
-                      }
+                    let textSubscription: Subscription;
+
+                    interruptStreaming$
+                      .pipe(
+                        take(1),
+                        tap(() => {
+                          streamingTextStatus.next(
+                            plainToInstance(FailedMessageStatus, { reason: "Interrupted" }),
+                          );
+                          stopStreamingText();
+                          textSubscription.unsubscribe();
+                        }),
+                      )
+                      .subscribe();
+
+                    textSubscription = textMessageContentStream.subscribe({
+                      next: async (e: RuntimeEvent) => {
+                        if (e.type == RuntimeEventTypes.TextMessageContent) {
```

Comment:
Is it possible we end up getting a next event but we called `stopStreamingText()` before?

Reply:
I don't think so.